tests/containers/state/StateTimeVisualizationSectionContainer-test.jsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * StateTimeVisualizationSectionContainer-test.jsx
+ * Created by David Trinh 5/18/18
+ */
+
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+
+import { StateTimeVisualizationSectionContainer } from
+    'containers/state/StateTimeVisualizationSectionContainer';
+
+import { mockActions, mockRedux, mockYears, mockQuarters, mockMonths } from './mockData';
+
+jest.mock('helpers/searchHelper', () => require('./mockSearchHelper'));
+
+// mock the child component by replacing it with a function that returns a null element
+jest.mock('components/state/spendingovertime/StateTimeVisualizationSection', () => jest.fn(() => null));
+
+
+jest.mock('components/search/visualizations/time/TimeVisualizationSection', () =>
+    jest.fn(() => null));
+
+describe('StateTimeVisualizationSectionContainer', () => {
+    it('should make an API call for the selected state on mount', async () => {
+        const container = mount(<StateTimeVisualizationSectionContainer
+            {...mockRedux}
+            {...mockActions} />);
+
+        const fetchData = jest.fn();
+        container.instance().fetchData = fetchData;
+
+        container.instance().componentDidMount();
+        await container.instance().apiRequest.promise;
+
+        expect(fetchData).toHaveBeenCalledTimes(1);
+    });
+
+    it('should make an API call when the fiscal year changes', async () => {
+        const container = mount(<StateTimeVisualizationSectionContainer
+            {...mockRedux}
+            {...mockActions} />);
+
+        const fetchData = jest.fn();
+        container.instance().fetchData = fetchData;
+
+        const stateProfile = Object.assign({}, mockRedux.stateProfile, {
+            fy: '2018'
+        });
+
+        const nextProps = Object.assign({}, mockRedux, {
+            stateProfile
+        });
+
+        container.setProps(nextProps);
+
+        await container.instance().apiRequest.promise;
+
+        expect(fetchData).toHaveBeenCalledTimes(1);
+    });
+
+    describe('parseData', () => {
```

Comment:
Can you write additional, specific unit tests for the `generateXXX` functions you're using inside `parseData` that are implemented by `StateTimeVisualizationSectionContainer`?